Tasks/AzureResourceManagerTemplateDeploymentV3/models/DeployAzureRG.ts

```diff
@@ -0,0 +1,94 @@
+import tl = require("azure-pipelines-task-lib/task");
+import msRestAzure = require('azure-arm-rest-v2/azure-arm-common');
+import { AzureRMEndpoint } from 'azure-arm-rest-v2/azure-arm-endpoint';
+
+export class AzureRGTaskParameters {
+
+    public action: string;
+    public resourceGroupName: string;
+    public location: string;
+    public csmFile: string;
+    public csmParametersFile: string;
+    public templateLocation: string;
+    public csmFileLink: string;
+    public csmParametersFileLink: string;
+    public overrideParameters: string;
+    public outputVariable: string;
+    public subscriptionId: string;
+    public endpointPortalUrl: string;
+    public deploymentName: string;
+    public deploymentMode: string;
+    public credentials: msRestAzure.ApplicationTokenCredentials;
+    public deploymentOutputs: string;
+    public addSpnToEnvironment: boolean;
+    public connectedService: string;
+    public deploymentScope: string;
+    public managementGroupId: string;
+
+    private async getARMCredentials(connectedService: string): Promise<msRestAzure.ApplicationTokenCredentials> {
+        var azureEndpoint = await new AzureRMEndpoint(connectedService).getEndpoint();
+        return azureEndpoint.applicationTokenCredentials;
+    }
+
+    public async getAzureRGTaskParameters() : Promise<AzureRGTaskParameters>
+    {
+        try {
+
+            //Deployment Scope
+            this.deploymentScope = tl.getInput("deploymentScope");
+            if(!this.deploymentScope){
+                this.deploymentScope = "Resource Group";
+            }
+
+            this.connectedService = tl.getInput("ConnectedServiceName", true);
+            var endpointTelemetry = '{"endpointId":"' + this.connectedService + '"}';
+            console.log("##vso[telemetry.publish area=TaskEndpointId;feature=AzureResourceManagerTemplateDeployment]" + endpointTelemetry);
+            this.endpointPortalUrl = tl.getEndpointDataParameter(this.connectedService, "armManagementPortalUrl", true);
+            this.action = tl.getInput("action");
+
+            //Management Group Id
+            if(this.deploymentScope === "Management Group"){
+                this.managementGroupId = tl.getEndpointDataParameter(this.connectedService, "ManagementGroupId", false);
+            }
```

Comment:
we need to check if management group id turns out to be null/empty and it needs to be used in controller. I think we should do a check when creating controller and throw there

Reply:
Similar check if required for subscription id and resource group as well

Reply:
resource group check we already doing in line no 63

Reply:
getEndpointDataParameter does this job for us, as I've marked this parameter as non-optional. SO it will throw if value is null or not present